src/tpm2-tss-engine-ecc.c

```diff
@@ -54,6 +57,23 @@ EC_KEY_METHOD *ecc_methods = NULL;
 #ifdef HAVE_OPENSSL_DIGEST_SIGN
 static int (*ecdsa_pkey_orig_copy)(EVP_PKEY_CTX *dst, const EVP_PKEY_CTX *src);
 static void (*ecdsa_pkey_orig_cleanup)(EVP_PKEY_CTX *ctx);
+static int (*sm2_digest_custom_default) (EVP_PKEY_CTX *ctx, EVP_MD_CTX *mctx) = NULL;
+static int (*pkey_sm2_sign_default) (EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
+                         const unsigned char *tbs, size_t tbslen) = NULL;
+static int (*sm2_sign_init_default)(EVP_PKEY_CTX*) = NULL;
+static int (*sm2_ctrl_default)(EVP_PKEY_CTX*, int, int, void*) = NULL;
+static int (*sm2_ctrl_str_default)(EVP_PKEY_CTX*, const char*, const char*) = NULL;
+
+/* The default user id as specified in GM/T 0009-2012.
+ * For details, please refer to GM/T 0003.1-2012 -- 0003.5-2012 SM2 and
+ * described at http://www.gmbz.org.cn/upload/2024-11-18/1731899635265027729.pdf
+ */
+#define SM2_DEFAULT_USERID "1234567812345678"
+#define SM2_DEFAULT_USERID_LEN (sizeof(SM2_DEFAULT_USERID) - 1)
+
+const EC_KEY *ec_sm2 = NULL;
+const EVP_PKEY *pkey_sm2 = NULL;
```

Comment:
Add documentation on what these global variables mean. Something like:

ec_sm2 caches the last seen TPM2 SM2 public key. The cache is only valid when the EVT_PKEY structure matches pkey_sm2. The cache is needed since pkey_sm2_sign needs to get ahold of the actual key which is inaccessible inside the EVP_PKEY is an opaque struct.